### PR TITLE
304 call mocks are not matching call signatures when one or more program parameters are qualified

### PR DESCRIPTION
**Pull request by ThomasNellemann**
CALL MOCK and VERIFY CALL will read parameters with qualifiers. 

When tests are passed and COBOL is generated, the mocking and verifying is done correctly for CALLs with and without parameters that are written with qualifiers.

### CODE REVIEW

File: src/test/java/org/openmainframeproject/cobolcheck/TestSuiteParserParsingTest.java [hunk 1]
```diff
@@ -506,10 +506,146 @@ public void verify_can_attach_to_call_mock_with_arguments() {
...
+        expectedResult.add("           PERFORM UT-ASSERT-ACCESSES");
+        try {
+            testSuiteParser.getParsedTestSuiteLines(
+                    new BufferedReader(new StringReader(testSuite.toString())),numericFields);
+            List<String> actualResult = new ArrayList<>();
+
+            testSuiteParser.handleEndOfVerifyStatement(actualResult, Constants.EMPTY_STRING, Constants.EMPTY_STRING);
+            // the following assert must not be performed!
+            fail("An exception should have been thrown :(");
+        } catch (VerifyReferencesNonexistentMockException mockException) {
```
Comment (dakaa16) [inline src/test/java/org/openmainframeproject/cobolcheck/TestSuiteParserParsingTest.java:572]: Junit has a built in function for testing if the right exception has been thrown: assertThrows(...). You can see the test  `it_throws_when_config_file_name_is_null()` for reference

**Review by dakaa16 — DISMISSED** [review]
I would like some more tests about the arguments. See the test: `call_mock_gets_correct_arguments()` as an example. A test should exist for arguments with one qualified reference and one with multiple `THIS IN SOMETHING IN SOMETHING-ELSE IN ...`.

I would also like a test for loading in the arguments from the InterpreterController. See `it_sets_possible_mock_args()` as an example.

Otherwise it looks good 😄

File: testsuites/concatenatedTestsuites
Comment (Rune-Christensen) [inline testsuites/concatenatedTestsuites:1]: @dakaa16 
Should this file be listet in .gitignore, and removed from the repository?

File: src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java [hunk 2]
```diff
@@ -557,24 +578,23 @@ private static Integer determineCobolLevelNumber(String levelNumberString){
...
 
-    private static String[] extractStatementWords(List<CobolLine> currentStatement){
-    String statementString = "";
-    boolean[] isContinuationLine = new boolean[currentStatement.size()];
-    Arrays.fill(isContinuationLine, false);
-    for(int i = 1; i < currentStatement.size(); ++i) {
+    private static String[] extractStatementWords(List<CobolLine> currentStatement) {
+        String statementString = "";
+        boolean[] isContinuationLine = new boolean[currentStatement.size()];
+    Arrays.fill(isContinuationLine, false);for (int i = 1; i < currentStatement.size(); ++i) {
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java:584]: I would like the for loop, to be on its own line.

File: src/main/java/org/openmainframeproject/cobolcheck/features/testSuiteParser/TestSuiteParser.java [hunk 3]
```diff
@@ -202,7 +206,10 @@ public List<String> getParsedTestSuiteLines(BufferedReader testSuiteReader,
                 // to next token
                 expectMockArguments = false;
                 expectUsing = false;
-                handleEndOfMockStatement(testSuiteReader, testSuiteToken, false);
+
+                handleEndOfMockStatement(testSuiteReader, testSuiteToken, false, summizedTokensForArgument, parseDataUsingForSummizedTokens);
+                summizedTokensForArgument =Constants.EMPTY_STRING;
+                parseDataUsingForSummizedTokens =Constants.EMPTY_STRING;
```
[the remark this answers is not in the document]
Comment (ThomasNellemann) [reply src/main/java/org/openmainframeproject/cobolcheck/features/testSuiteParser/TestSuiteParser.java]: new class introduced to encapsulate the values

**Review by Rune-Christensen — APPROVED** [review]

**Pull request merged**